tests/functional/smoke/modes.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2021-2022 Micron Technology, Inc. All rights reserved.
+
+#doc: test behavior of kvdb open modes (rdonly, rdonly_replay, diag, rdwr)
+
+. common.subr
+
+trap kvdb_drop EXIT
+kvdb_create
+
+props="-oinodesc=3,datac=7"
+
+# Load 8K records
+kvt -i8k -cv -m1 "$props" "$home"
+
+# PUTs must fail on a KVDB opened in the following modes
+modes="rdonly rdonly_replay diag"
+for mode in $modes
+do
+    cmd -e kvt -T10,4 -l8 -m1 "$props" "$home" kvdb-oparms mode="$mode"
+done
```

Comment:
I prefer moving away from smoke tests where possible.  It puts too much code (e.g., the kvt app) between the test and what is being tested.  Can this be implemented in `tests/functional/api/kvdb_api_test.c`?

Reply:
I tend to agree.

Reply:
The smoke test verifies that KVDBs can be opened and queried in a read only mode when the underlying files are read only.  That might be hard to do in the API test suite -- but that is also not an API test.  So maybe leave those types of test here and do the testing of the `hse_kvdb_open` API in the API test.

Reply:
Will do!